### ISSUE TITLE
Exchanging two items causes two moves

### ISSUE THREAD

**Issue by georgbachmann** [Instagram/IGListKit#1302]
## New issue checklist

- [x] I have reviewed the [`README`](https://github.com/Instagram/IGListKit/blob/master/README.md) and [documentation](http://instagram.github.io/IGListKit)
- [x] I have searched [existing issues](https://github.com/Instagram/IGListKit/issues) and this is not a duplicate
- [x] I have attempted to reproduce the issue and include an example project.

### General information

- `IGListKit` version: 3.4.0
- iOS version(s): iOS 12.1.3
- CocoaPods/Carthage version: Cocoapods 1.6.0.beta.1
- Xcode version: 10.1 (10B61)
- Devices/Simulators affected: Any
- Reproducible in the demo project? (Yes/No): Yes
- Related issues: Non found

### Debug information

I am wondering the following. I have a list with in my test two items. [0, 1]. Then I rearrange them like so: [1, 0]. I'd expect to get ONE move from the `ListDiff`, but instead I get two?
I am wondering if I do something wrong or if this is intended behaviour? I want to use `ListDiff` for a `UITableView`. Cause if I'd apply both changes, they'd cancel each other out.

This is how my model looks:
```final class TestViewModel: NSObject, ListDiffable {
    let identifier: Int
    let sortOrder: Int
    
    init(identifier: Int, sortOrder: Int) {
        self.identifier = identifier
        self.sortOrder = sortOrder
    }
    
    public func diffIdentifier() -> NSObjectProtocol {
        return NSNumber(value: identifier)
    }
    
    public func isEqual(toDiffableObject object: ListDiffable?) -> Bool {
        
        guard let otherObject = (object as? TestViewModel) else { return false }
        
        return identifier == otherObject.identifier &&
               sortOrder == otherObject.sortOrder
    }
}
```

So now I have my data before the change
```
        let oldData = [
            TestViewModel(identifier: 0, sortOrder: 0),
            TestViewModel(identifier: 1, sortOrder: 1),
            ]```


and this will be the data after the change


```        let newData = [
            TestViewModel(identifier: 1, sortOrder: 0),
            TestViewModel(identifier: 0, sortOrder: 1),
            ]```


So basically I just changed the sortOrder.
Now I make a diff like so:

```
let diff = ListDiff(oldArray: oldData, newArray: newData, option: .equality)```

I'd expect there to be two changes because of the sortOrder... Which is exactly what happens! Good!
But I also have two moves.

```(lldb) po diff.moves
▿ 2 elements
  - 0 : <IGListMoveIndex 0x2831d3520; from: 1; to: 0;>
  - 1 : <IGListMoveIndex 0x2831d3580; from: 0; to: 1;>```

Which is wired. Cause moving item 1 to 0 is everything needed, isn't it?

I want to use the `ListDiff` for my `UITableView` but if I'd move 1 -> 0 and then 0 -> 1 the two moves basically cancel each other out?!?
